core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableValues.java

```diff
@@ -65,9 +65,20 @@ public static EnumerableValues create(RelOptCluster cluster,
     return new EnumerableValues(cluster, rowType, tuples, traitSet);
   }
 
+  /** Creates an EnumerableValues. */
+  public static EnumerableValues create(Values input) {
+    final RelOptCluster cluster = input.getCluster();
```

Comment:
How about we modify the other `create` to support an explicit `traitSet` param ?

Reply:
I think the `copy` method should be handling that scenario